### PR TITLE
LPS-123368 Create upload field to support the file upload by Guest user

### PR DESCRIPTION
**Pull request by alinedoleron**
Related Issue: https://issues.liferay.com/browse/LPS-123368

### CODE REVIEW

**Comment by liferay-forms** [comment]
Hi, you don’t need to run the CI commands anymore, I’ll do it for you and let you know if any issue happens.

**Comment by liferay-forms** [comment]
ci:test:sf

**Comment by liferay-forms** [comment]
ci:test:relevant

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/DocumentLibrary/DocumentLibrary.es.js [hunk 1]
```diff
@@ -12,11 +12,16 @@
...
  */
 
+import './DocumentLibrary.scss';
+
 import ClayButton, {ClayButtonWithIcon} from '@clayui/button';
 import ClayCard from '@clayui/card';
 import {ClayInput} from '@clayui/form';
 import ClayIcon from '@clayui/icon';
+import ClayProgressBar from '@clayui/progress-bar';
+import axios from 'axios';
```
Comment (natocesarrego) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/DocumentLibrary/DocumentLibrary.es.js:22]: @alinedoleron, can we add the import following the alphabetic order? This should be on line 17, except if SF tells us to not do it.

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/DocumentLibrary/DocumentLibrary.es.js [hunk 2]
```diff
@@ -12,11 +12,16 @@
...
 import {usePage} from 'dynamic-data-mapping-form-renderer';
+import {convertToFormData} from 'dynamic-data-mapping-form-renderer/js/util/fetch.es';
```
Comment (natocesarrego) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/DocumentLibrary/DocumentLibrary.es.js:24]: @alinedoleron, can we add the import following the alphabetic order? This should be on line 22, except if SF tells us to not do it.

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/DocumentLibrary/DocumentLibrary.es.js [hunk 3]
```diff
@@ -202,9 +207,83 @@ const DocumentLibrary = ({
 	);
 };
 
+const Upload = ({
```
Comment (natocesarrego) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/DocumentLibrary/DocumentLibrary.es.js:210]: @alinedoleron, how about name it to `GuestUploadFile` to make more explicit the purpose of this component?

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/DocumentLibrary/DocumentLibrary.es.js [hunk 4]
```diff
@@ -280,6 +364,56 @@ const Main = ({
 		itemSelectorDialog.open();
 	};
 
+	const configureErrorMessage = (message) => {
+		const enable = message ? true : false;
+
+		setErrorMessage(message);
```
Comment (natocesarrego) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/DocumentLibrary/DocumentLibrary.es.js:370]: @alinedoleron, please declare the variables just before its use. 😉 
```
	const configureErrorMessage = (message) => {
		setErrorMessage(message);

		const enable = message ? true : false;

		setDisplayErrors(enable);
		setValid(!enable);
	};
```

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/DocumentLibrary/DocumentLibrary.es.js [hunk 5]
```diff
@@ -280,6 +364,56 @@ const Main = ({
...
+					configureErrorMessage(error.message);
+					setCurrentValue(null);
+					onChange(event, '{}');
+				}
+				else {
+					configureErrorMessage('');
+					setCurrentValue(JSON.stringify(file));
+					onChange(event, JSON.stringify(file));
+				}
+				setProgress(0);
```
Comment (natocesarrego) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/DocumentLibrary/DocumentLibrary.es.js:413]: @alinedoleron, please add line breakers among these lines to separate the methods invocations. Please check how you did it on line 391 until line 395. Let's make it consistent. Keep it in mind. 🙂

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/DocumentLibrary/DocumentLibrary.es.js [hunk 6]
```diff
@@ -292,26 +426,45 @@ const Main = ({
...
-			/>
+			{allowGuestUsers && !isSignedIn ? (
+				<Upload
+					fileEntryTitle={fileEntryTitle}
+					fileEntryURL={fileEntryURL}
+					id={id}
+					name={name}
+					onClearButtonClicked={(event) => {
+						setCurrentValue(null);
+						onChange(event, '{}');
```
Comment (natocesarrego) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/DocumentLibrary/DocumentLibrary.es.js:437]: @alinedoleron, please check previous comment.

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/DocumentLibrary/DocumentLibrary.scss [hunk 7]
```diff
@@ -0,0 +1,20 @@
+.liferay-ddm-form-field-document-library {
```
Comment (natocesarrego) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/DocumentLibrary/DocumentLibrary.scss:1]: @alinedoleron, a CSS class with the same name already exists [here](https://github.com/liferay/liferay-portal/blob/aeadbfb6f2563a86deb75befd0ecd3b1dd2284eb/modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-renderer/src/main/resources/META-INF/resources/css/main.scss#L220-L224), should we keep both? Can we merge it to a single place? Maybe we can keep this class on the existent **main.scss**.

**Review by natocesarrego — CHANGES_REQUESTED** [review]

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/DocumentLibrary/DocumentLibrary.es.js (hunk 1 above)
Comment (alinedoleron) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/DocumentLibrary/DocumentLibrary.es.js:22]: I suspect this but I thought SF would fix it. I'll make the change

Comment (alinedoleron) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/DocumentLibrary/DocumentLibrary.es.js:22]: Ops, now when I tried the line got red and I used the the Fix auto-fixable problems tools I it replaced to the previous place.

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/DocumentLibrary/DocumentLibrary.es.js (hunk 2 above)
Comment (alinedoleron) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/DocumentLibrary/DocumentLibrary.es.js:24]: I realized the order is made according to the alphabetic order after "from".

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/DocumentLibrary/DocumentLibrary.es.js (hunk 4 above)
Comment (alinedoleron) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/DocumentLibrary/DocumentLibrary.es.js:370]: Sorry, I didn't get your point here.
Isn't the enable variable before the usage yet?

**Comment by alinedoleron** [comment]
new pr https://github.com/natocesarrego/liferay-portal/pull/921

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/DocumentLibrary/DocumentLibrary.es.js (hunk 1 above)
Comment (natocesarrego) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/DocumentLibrary/DocumentLibrary.es.js:22]: Let's keep where they are.

**Pull request closed without merging**